File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [0] * n
    graph = [[] for _ in range(n)]
        
    def dfs(v):
        visited[v] = True
        for i in range(n):
            if i != n and computers[v][i] == 1:
                if not visited[i]:
                    dfs(i)
        
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer += 1
        
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    void dfs(int v, int[][] computers, boolean[] visited) {
        visited[v] = true;
        for(int i=0; i<computers.length; i++) {
            if(i != v && computers[v][i] == 1 && !visited[i]) {
                dfs(i, computers, visited);
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[computers.length];
    
        
        for(int i=0; i<computers.length; i++) {
            if(!visited[i]) {
                dfs(i, computers, visited);
                answer += 1;
            }
        }
        
        
        
        return answer;
    }
}